opossum-tool/OpossumUI
# Use worker for top progress bar

### Summary of changes

Adapt the worker for the folder progress bar and use it for the top progress bar.

### Context and reason for change

See https://github.com/opossum-tool/OpossumUI/issues/1110

This PR aims to speed up saving. The following two functions were identified as taking up most of the time when saving a large file:
1. `getUpdatedProgressBarData` (~1300ms on my setup)
2. `saveManualAndResolvedAttributionsToFile` (~300ms on my setup)
`getUpdatedProgressBarData` is a function that calculates data for the progress bar at the top of the UI. Since it takes up a lot of time and doesn't need to be blocking, it was moved to a worker where it can be executed asynchronously.

### How can the changes be tested

Save changes to a large OpossumUI input file. Use the Performance tab to check how long it takes (or just feel the difference).

From my test (~600,000 files)
Before: 1.55s
After: 0.51s
